test/test_core.py

```diff
@@ -3958,6 +3958,48 @@ def test_dlfcn_rtld_local(self):
 
     self.do_runf('main.c', 'main\nfunc_a\nfunc_sub\nfunc_b\nfunc_sub\ndone\n')
 
+  def test_dlfcn_with_preload(self):
+    # Create chain of dependencies and load the first libary with preload plugin.
+    # main -> libB.so -> libA.sp
+    create_file('libA.c', r'''
+      #include <stdio.h>
+      int libA_fun() {
+        return 23;
+      }
+    ''')
+    self.run_process([EMCC, 'libA.c', '-sSIDE_MODULE', '-o', 'libA.so'])
+    create_file('libB.c', r'''
+      #include <stdio.h>
+      int libA_fun();
+
+      int libB_fun() {
+        return libA_fun()*2;
+      }
+    ''')
+    self.run_process([EMCC, 'libB.c', '-sSIDE_MODULE', '-o', 'libB.so', 'libA.so'])
+    create_file('main.c', r'''
+      #include <assert.h>
+      #include <dlfcn.h>
+      #include <stdio.h>
+      #include <sys/stat.h>
+
+      int main() {
+        // Check the file exists in the VFS
+        struct stat statbuf;
+        assert(stat("/libB.so", &statbuf) == 0);
+        void *lib_handle = dlopen("/libB.so", RTLD_NOW);
+        assert(lib_handle);
+        typedef int (*intfunc)();
+        intfunc x = (intfunc)dlsym(lib_handle, "libB_fun");
```

Comment:
How does this test fail without this change?

Reply:
It fails with this error:

`Assertion failed: undefined symbol 'libA_fun'. perhaps a side module was not linked in? if this global was expected to arrive from a system library, try to build the MAIN_MODULE with EMCC_FORCE_STDLIBS=1 in the environment`